src/components/Vault/BasicVault/VaiCard.tsx

```diff
@@ -95,68 +90,43 @@ function VaultCard({ settings }: VaultCardProps) {
-                  .toString(10),
-              )}{' '}
-              XVS
-            </div>
-          </Col>
-          <Col className="col-item" lg={{ span: 4 }} md={{ span: 6 }} xs={{ span: 12 }}>
-            <div className="title">VAI Staking APR</div>
-            <div className="content">{settings.vaiAPY}%</div>
-          </Col>
-          <Col className="col-item" lg={{ span: 4 }} md={{ span: 6 }} xs={{ span: 12 }}>
-            <div className="title">Total VAI Staked</div>
-            <div className="content">
-              {settings.vaultVaiStaked
-                ? commaFormatter(new BigNumber(settings.vaultVaiStaked).dp(4, 1).toString(10))
-                : 0}{' '}
-              VAI
-            </div>
-          </Col>
-          <Col className="col-item" lg={{ span: 4 }} md={{ span: 6 }} xs={{ span: 12 }}>
-            <div className="title">XVS Daily Emission</div>
-            <div className="content">{commaFormatter(dailyEmission.toString(10))} XVS</div>
-          </Col>
-          <Col
-            className="col-item expand-icon-wrapper"
-            lg={{ span: 2 }}
-            xs={{ span: 24 }}
-            onClick={() => setExpanded(!expanded)}
-          >
-            <img className="expand-icon" alt="open" src={arrowDownImg} />
-          </Col>
-        </Row>
-      </div>
+      <CardHeader
+        stakedToken="VAI"
+        rewardToken="XVS"
+        apy={settings.vaiAPY || 0}
+        totalStakedAmount={new BigNumber(settings.vaultVaiStaked || 0)}
+        totalPendingRewards={totalPendingRewards.div(1e18)}
+        dailyEmission={dailyEmission}
+        onExpand={() => {
+          setExpanded(!expanded);
+        }}
+      />
       <div className="content-container">
         {expanded ? (
-          <VAICardContent
+          <CardContent
             userPendingReward={userPendingReward}
-            userVaiBalance={userVaiBalance}
-            userVaiAllowance={userVaiAllowance}
-            userVaiStakedAmount={userVaiStakedAmount}
+            userStakedTokenBalance={userVaiBalance}
+            userStakedAllowance={userVaiAllowance}
+            userStakedAmount={userVaiStakedAmount}
+            stakedToken="VAI"
+            rewardToken="XVS"
+            onClaimReward={async () => vaiVaultContract.methods.claim().send({ from: account })}
+            onStake={async (stakeAmount) => vaiVaultContract.methods
+              .deposit(
+                stakeAmount.toFixed(0),
+              )
+              .send({ from: account })}
+            onApprove={async (amt) => vaiTokenContract.methods
+              .approve(
+                vaiVaultContract.options.address,
+                amt.toFixed(10),
+              )
+              .send({ from: account })}
+            onWithdraw={async (amt) => vaiVaultContract.methods
+              .withdraw(
+                amt.toFixed(0),
+              )
+              .send({ from: account })}
           />
         ) : null}
```

Comment:
```suggestion
        )}
```
... this way you don't need to return `null` if `expanded` is false, it will have the same behaviour :)